xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpSession.java

```diff
@@ -804,26 +822,19 @@ private enum OveractivityType {
     /**
      * Check that the client SHOULD NOT make more simultaneous requests than specified
      * by the 'requests' attribute in the connection manager's Session Creation Response.
-     * However the client MAY make one additional request if it is to pause or terminate a session.
+     * However, the client MAY make one additional request if it is to pause or terminate a session.
      *
      * @see <a href="http://www.xmpp.org/extensions/xep-0124.html#overactive">overactive</a>
      * @param connection the new connection.
      * @throws HttpBindException if the connection has violated a facet of the HTTP binding
      *         protocol.
      */
     private void checkOveractivity(HttpConnection connection) throws HttpBindException {
-        int pendingConnections = 0;
+        int pendingConnections;
         OveractivityType overactivity = OveractivityType.NONE;
 
         synchronized (connectionQueue) {
```

Comment:
Not sure this `synchronized` is necessary, given the scope of the containing block?

Reply:
You might be right. As I'm working on this PR, my plan was to incrementally progress to a point where we can hopefully significantly reduce the usage of locks. Up until that time, I've not yet spent to much thought-cycles on evaluating the existing locks, as that's very much on my to-do list for this PR. Work in progress!